Objects/longobject.c

```diff
@@ -5421,6 +5432,59 @@ int_bit_length_impl(PyObject *self)
     return NULL;
 }
 
+/*[clinic input]
+int.bit_count
+
+Number of ones in the binary representation of self.
+
+>>> bin(13)
+'0b1101'
+>>> (13).bit_count()
+3
+[clinic start generated code]*/
+
+static PyObject *
+int_bit_count_impl(PyObject *self)
+/*[clinic end generated code: output=2e571970daf1e5c3 input=a428900d3e39a606]*/
+{
```

Comment:
Can we use a `PyLongObject *self` variable here, and rename the parameter to "self_obj", to avoid having to cast self to `PyLongObject *` multiple times? In pratice, it doesn't seem easy to do, so it's also ok to keep the code as it is :-)

Reply:
I reworked the code a bit to reduce the number of casts; most names now have type `PyObject *` rather than `PyLongObject *`.